### PR TITLE
Added 3 'Notes for Professionals' books

### PR DESCRIPTION
**Pull request by iaminebriki**
## What does this PR do?
Add some 'Notes for Professionals' books

### Description
The added books are:
- Linux Notes for Professionals
- Entity Framework Notes for Professionals
- Hibernate Notes for Professionals
### Why is this valuable (or not)?
It extend the books list with more free resouces!
### How do we know it's really free?
The 'Notes for Professionals' book series are compiled from StackOverflow documentation and are completely free to download, you can check their website: [books.goalkicker.com](https://books.goalkicker.com/)
### For book lists, is it a book? For course lists, is it a course? etc.
Yes, it's 3 different books
## Checklist:
- [X] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [X] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [X] Include author(s) and platform where appropriate.
- [X] Put lists in alphabetical order, correct spacing.
- [X] Add needed indications (PDF, access notes, under construction).
- [X] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by iaminebriki** [comment]
Kindly add `hacktoberfest-accepted` label

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -708,6 +708,7 @@ Books that cover a specific programming language can be found in the [BY PROGRAM
 * [How to Make a Computer Operating System](https://github.com/SamyPesse/How-to-Make-a-Computer-Operating-System) - Samy Pesse (:construction: *in process*)
 * [How to write a simple operating system in assembly language](https://mikeos.sourceforge.net/write-your-own-os.html) - Mike Saunders (HTML)
 * [Linux From Scratch](https://www.linuxfromscratch.org/lfs/view/stable/) - Gerard Beekmans, Bruce Dubbs, Ken Moffat, Pierre Labastie, et al. (HTML, [PDF, downloads...](https://www.linuxfromscratch.org/lfs/downloads/stable/))
+* [Linux Notes for Professionals](https://books.goalkicker.com/LinuxBook) - Compiled from StackOverflow Documentation (PDF)
```
Comment (eshellman) [inline books/free-programming-books-subjects.md:711]: can't list it in two places, not withstanding the link above it.

Comment (iaminebriki) [reply books/free-programming-books-subjects.md]: I have removed the linux book from the "per subjects" list

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**